gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceId.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.gcloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents a Google Cloud Resource Manager Resource ID
```

Comment:
See my comment regarding `ProjectInfo.Builder.setParent`. If this is not yet support maybe we should not have it (and instead create an issue to support it when available and until then just
keep it internally). If it is available, then we need more docs. e.g. what is the id and what are the values for type...